services/settings_service.py

```diff
@@ -355,9 +357,9 @@ async def set_wingman_pro_settings(
                 toast=ToastType.NORMAL,
                 color=LogType.POSITIVE,
             )
-            if old_va_threshold != va_energy_threshold:
+            if old_va_threshold != va_energy_threshold or old_stt_provider != stt_provider:
                 await self.settings_events.publish(
-                    "va_treshold_changed", va_energy_threshold
+                    "va_settings_changed", va_energy_threshold
```

Comment:
for consistency we should pass `stt_provider` as event args, too? Maybe a tuple or dict of event args.